paddle/platform/for_range.h

```diff
@@ -64,15 +64,17 @@ struct ForRange<CUDADeviceContext> {
   inline void operator()(Function func) const {
     constexpr int num_threads = 1024;
     int block_size = limit_ <= num_threads ? limit_ : num_threads;
-    int grid_size = (limit_ + num_threads - 1) / num_threads;
-
-    if (grid_size == 1) {
-      ForRangeElemwiseOpGridIsOne<<<1, block_size, 0, dev_ctx_.stream()>>>(
-          func);
-    } else {
-      ForRangeElemwiseOp<<<grid_size, block_size, 0, dev_ctx_.stream()>>>(
-          func, limit_);
+
+    if (block_size < 1024) {
+      int size = 1;
+      while (size < block_size) size <<= 1;
+      block_size = size;
```

Comment:
1. The warp size is 32 in Nvidia GPU, if limit_ is less than 1024,  make the block size be divided by 32 by the following calculation:

```c++
if (block_size < 1024) {
    block_size = ((blokc_size + 31) >> 5) << 5;
}

``` 

2. The following function from line 44 to line 48 also needs to be updated. 

```c++
template <typename Function>
__global__ static void ForRangeElemwiseOpGridIsOne(Function func) {
  size_t idx = static_cast<size_t>(threadIdx.x);
  func(idx);
}
```

Reply:
I am not sure this optimization is necessary.

If the number of thread cannot be divided by wrap size, there will be inactived threads. However, even we make the number of thread can be divided by wrap size, there will be inactived threads, too. Because all threads run the same code, there will be some thread not fulfill the `if` statement and wait other threads.

Reply:
@reyoung It seems reasonable, but I have not seen any document directing uses to set block_size in this way.
I don't do benchmark, so I don't tell you which is better.